### PR TITLE
(maint) Bump jruby hocon to 1.3.1

### PR DESCRIPTION
**Pull request by Iristyle**
- Includes feature from 1.2.7 to load environment variables into lists
   like:

    FOO.1
    FOO.2

    See https://github.com/puppetlabs/ruby-hocon/pull/117

### CODE REVIEW

**Comment by MikaelSmith** [comment]
Oh, we do use ruby hocon here. Probably need both.

**Comment by Iristyle** [comment]
I'm not sure all the places things will need to be updated, nor do I know yet where `puppetlabs/typesafe-config` is referenced

File: project.clj [hunk 1]
```diff
@@ -27,7 +27,7 @@
 
   :min-lein-version "2.9.1"
 
-  :parent-project {:coords [puppetlabs/clj-parent "4.6.10"]
+  :parent-project {:coords [puppetlabs/clj-parent "4.6.11"]
```
Comment (Magisus) [inline project.clj:30]: This update will be automatic when clj-parent is released. However, we will need to do the bump by hand in pe-pse.

**Comment by Magisus** [comment]
This can be rebased now, the clj-parent bump is in 6.x.

File: project.clj (hunk 1 above)
Comment (Iristyle) [reply project.clj:30]: 👍 

PR is at https://github.com/puppetlabs/pe-puppet-server-extensions/pull/1262 

I'll update this PR once clj-parent ships and after this repo is auto-updated

**Review by mwaggett — APPROVED** [review]

**Comment by Iristyle** [comment]
Looks like we have some failing specs

```
FAIL in (cert-decoding-failures) (request_handler_core_test.clj:267)
Improperly URL encoded content
expected: (thrown+? [:kind :bad-request :msg (str "Unable to URL decode the x-client-cert header: " "For input string: \"1%\"")] (jruby-request-with-client-cert-header "%1%2"))
  actual: "thrown+?: {:kind :bad-request, :msg \"No certs found in PEM read from x-client-cert\"} did not match [:kind :bad-request :msg (str \"Unable to URL decode the x-client-cert header: \" \"For input string: \\\"1%\\\"\")]"
```

**Comment by Iristyle** [comment]
Yeah, so the clj-parent 4.6.10 to 4.6.11 bump didn't pass at https://github.com/puppetlabs/puppetserver/runs/1743672108 - 41f70ff addresses that

**Comment by Magisus** [comment]
For reference, the commit that broke us is https://github.com/ring-clojure/ring-codec/commit/780088590b57a0caf2d64f378f8ab852547a698e

**Review by mwaggett — APPROVED** [review]

**Pull request merged**